lib/AST/ASTImporter.cpp

```diff
@@ -4575,14 +4589,36 @@ Decl *ASTNodeImporter::VisitClassTemplateSpecializationDecl(
   if (FoundDef) {
     // We already have a class template specialization with these template
     // arguments.
-    
-    // FIXME: Check for specialization vs. instantiation errors.
-    if (!D->isCompleteDefinition() || IsStructuralMatch(D, FoundDef)) {
-      // The record types structurally match, or the "from" translation
-      // unit only had a forward declaration anyway; call it the same
-      // function.
+
+    if (!D->isCompleteDefinition()) {
+      // The "From" translation unit only had a forward declaration; call it
+      // the same declaration.
       return Importer.MapImported(D, FoundDef);
     }
+
+    if (IsStructuralMatch(D, FoundDef)) {
+
+      Importer.MapImported(D, FoundDef);
+
+      // Check and merge those fields which have been instantiated
+      // in the "From" context, but not in the "To" context.
+      for (auto *FromField : D->fields()) {
+        Importer.Import(FromField);
```

Comment:
There can be some check for nullptr result of the import, at least a warning (but return nullptr is not possible here).

Reply:
It is not exactly clear for me, what could we benefit from a warning?
Either the field is already in the "To" context, then `Import` will not return a nullptr, or it is getting imported newly and that import may fail with a nullptr.
Perhaps we should think about to report an ODR violation if there is a new Field instead.

Reply:
If a field was not imported correctly it can cause errors later at analyze, a warning can give help at debugging. The nullptr is even returned if the field already exists but the initializer expression must be imported (this is the normal use case of this `Import` call?) and the import failed.
There are other places when nullptr is returned from `Import`, probably a warning should be made at every place (in all cases when the reason for returning nullptr is not another failed `Import` call).